src/plugins/NEMD/MettDeamon.cpp

```diff
@@ -0,0 +1,2048 @@
+		bool bIsInsideSubdomain = false;
+		domainDecomp->getBoundingBoxMinMax(global_simulation->getDomain(), bbMin, bbMax);
+		bIsInsideSubdomain = x > bbMin[0] && x < bbMax[0] && y > bbMin[1] && y < bbMax[1] && z > bbMin[2] && z < bbMax[2];
+
+		if(true == bIsInsideSubdomain)
+			_binVector.at(nSlabindex).push_back(m1);
+
+		componentid = m1.componentid();
+		// TODO: The following should be done by the addPartice method.
+		dcomponents.at(componentid).incNumMolecules();
+*/
+		_particleVector.push_back(mol);
+
+		// Print status message
+		unsigned long iph = _numMoleculesRead / 100;
+		if( iph != 0 && (i % iph) == 0 )
+			global_log->info() << "Finished reading molecules: " << i/iph << "%\r" << flush;
+	}
+
+	ifs.close();
+}
+
+void Reservoir::readFromFileBinaryHeader()
+{
+	DomainDecompBase& domainDecomp = global_simulation->domainDecomposition();
+	XMLfileUnits inp(_filepath.header);
+
+	if(false == inp.changecurrentnode("/mardyn")) {
+		global_log->error() << "Could not find root node /mardyn in XML input file." << endl;
+		global_log->fatal() << "Not a valid MarDyn XML input file." << endl;
+		Simulation::exit(1);
+	}
+
+	bool bInputOk = true;
+	double dCurrentTime = 0.;
+	double dBL[3];
+	uint64_t nNumMols;
+	std::string strMoleculeFormat;
+	bInputOk = bInputOk && inp.changecurrentnode("headerinfo");
+	bInputOk = bInputOk && inp.getNodeValue("time", dCurrentTime);
+	bInputOk = bInputOk && inp.getNodeValue("length/x", dBL[0] );
+	bInputOk = bInputOk && inp.getNodeValue("length/y", dBL[1] );
+	bInputOk = bInputOk && inp.getNodeValue("length/z", dBL[2] );
+	bInputOk = bInputOk && inp.getNodeValue("number", nNumMols);
+	bInputOk = bInputOk && inp.getNodeValue("format@type", strMoleculeFormat);
+	double dVolume = 1;
+	for(uint8_t di=0; di<3; ++di)
+	{
+		this->setBoxLength(di, dBL[di] );
+		dVolume *= dBL[di];
+	}
+	_numMoleculesRead = nNumMols;
+	this->setVolume(dVolume);
+	this->setDensity(0, nNumMols / dVolume);
+
+	if(false == bInputOk)
+	{
+		global_log->error() << "Content of file: '" << _filepath.header << "' corrupted! Program exit ..." << endl;
+		Simulation::exit(1);
+	}
+
+	if("ICRVQD" == strMoleculeFormat)
+		_nMoleculeFormat = ICRVQD;
+	else if("IRV" == strMoleculeFormat)
+		_nMoleculeFormat = IRV;
+	else if("ICRV" == strMoleculeFormat)
+		_nMoleculeFormat = ICRV;
+	else
+	{
+		global_log->error() << "Not a valid molecule format: " << strMoleculeFormat << ", program exit ..." << endl;
+		Simulation::exit(1);
+	}
+}
+
+void Reservoir::readFromFileBinary(DomainDecompBase* domainDecomp)
+{
+	global_log->info() << "Reservoir::readFromFileBinary(...)" << endl;
+	// read header
+	this->readFromFileBinaryHeader();
+
+#ifdef ENABLE_MPI
+	if(domainDecomp->getRank() == 0) {
+#endif
+	global_log->info() << "Opening phase space file " << _filepath.data << endl;
+	std::ifstream ifs;
+	ifs.open(_filepath.data.c_str(), ios::binary | ios::in);
+	if (!ifs.is_open()) {
+		global_log->error() << "Could not open phaseSpaceFile " << _file
```

Comment:
not released -> change to std::unique_ptr!

Reply:
What should be changed here? _filepath is not a pointer

Reply:
sorry, wrongly indicated, I meant _moleculeDataReader :)
am doing this change for you and a few more minor things.